src/EFCore.PG.NTS/Extensions/NpgsqlNetTopologySuiteDbContextOptionsBuilderExtensions.cs

```diff
@@ -0,0 +1,50 @@
+using GeoAPI;
+using GeoAPI.Geometries;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Npgsql;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure.Internal;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Utilities;
+
+namespace Microsoft.EntityFrameworkCore
+{
+    /// <summary>
+    /// NetTopologySuite specific extension methods for <see cref="NpgsqlDbContextOptionsBuilder"/>.
+    /// </summary>
+    // ReSharper disable once InconsistentNaming
+    public static class NpgsqlNetTopologySuiteDbContextOptionsBuilderExtensions
+    {
+        /// <summary>
+        /// Use NetTopologySuite to access SQL Server spatial data.
+        /// </summary>
+        /// <returns> The options builder so that further configuration can be chained. </returns>
+        public static NpgsqlDbContextOptionsBuilder UseNetTopologySuite(
+            [NotNull] this NpgsqlDbContextOptionsBuilder optionsBuilder,
+            ICoordinateSequenceFactory coordinateSequenceFactory = null,
+            IPrecisionModel precisionModel = null,
+            Ordinates handleOrdinates = Ordinates.None,
+            bool geographyAsDefault = false)
+        {
+            Check.NotNull(optionsBuilder, nameof(optionsBuilder));
+
+            // TODO: Do we really need this here or not? What does the SQL Server plugin do?
+            NetTopologySuiteBootstrapper.Bootstrap();
```

Comment:
I think this is where the SqlServer/Sqlite spatial implementation may actually have an advantage over Npgsql's.

With Npgsql, NTS is supported at the ADO level - which is great because you can easily use it without EF Core. However, unlike with EF Core there's no fancy dependency injection in ADO, so at least at the moment everything is done with the global NTS bootstrapping, which as you say isn't ideal.

Looking at this again, things seem a bit messy - we bootstrap NTS both at the EF Core level and at the ADO level. I'll try to take a look at it soon to make sure we're doing the right thing, but at least for this PR I guess I'll leave things as they are.

PS In theory we could implement EF Core for NTS much like SqlServer does - bypass the ADO support entirely, using it to read and write bytes, and have NTS take care of serialization/parsing of PostGIS wire data. But I'm not sure it's worth it.

Reply:
Removed the bootstrapping as this is supposed to be taken care of at the ADO level.

A more general conversation about how to manage the GeoAPI/NetTopologySuite service provider is in progress in https://github.com/NetTopologySuite/GeoAPI/issues/53.